torch/utils/flop_counter.py

```diff
@@ -244,18 +246,32 @@ def __init__(
         if isinstance(mods, torch.nn.Module):
             mods = [mods]
         self.mods = mods
-        if mods is not None:
-            for mod in mods:
-                prefix = type(mod).__name__
-                for name, module in dict(mod.named_modules()).items():
-                    if name == "":
-                        name = prefix
-                    else:
-                        name = ".".join([prefix, name])
-                    module.register_forward_pre_hook(self._enter_module(name))
-                    module.register_forward_hook(self._exit_module(name))
+        # Keys will include the modules in `mods` and their submodules
+        self._module_to_forward_hook_handles: Dict[nn.Module, _ForwardHookHandles] = {}
         self.flop_mapping = {**flop_mapping, **custom_mapping}
 
+    def _register_forward_hooks(self):
+        if self.mods is None:
+            return
+        for mod in self.mods:
+            prefix = type(mod).__name__
+            for name, module in dict(mod.named_modules()).items():
```

Comment:
minor: I copied this over, but I was wondering why not just `for name, module in mod.named_modules():`?

Reply:
Uhh... no particular reason I think.